pkg/dataobj/planner/physical/dataobjscan.go

```diff
@@ -0,0 +1,37 @@
+package physical
+
+type DataObjLocation string
+
+type Direction int8
+
+const (
+	Forward Direction = iota
```

Comment:
In general I think we should be careful to assume that the zero value of any enum type is always meaningful; I'm worried it could lead to bugs which are difficult to track down. 

For `Direction`, I think defaulting to Forward is probably reasonable (or perhaps even defaulting to Backward), but for types like `ExpressionType` I think we should reconsider and make the zero value invalid.

Reply:
Good point. I never really thought about zero-values